docs/user-guide/a1-06-spirv-target-specifics.md

```diff
@@ -0,0 +1,316 @@
+
++--------+-----------------+-----------------+-----------------------+------------------+------------------+
+|        |  32-bit integer | 64-bit integer  |      32-bit float     |  64-bit float    |   16-bit float   |
++--------+-----------------+-----------------+-----------------------+------------------+------------------+
+| HLSL   |   Yes (SM5.0)   |   Yes (SM6.6)   | Only bit-wise (SM6.6) |       No         |      No
+| GLSL   |   Yes (GL4.3)   | Yes (GL4.4+ext) |    Yes (GL4.6+ext)    | Yes (GL4.6+ext)  | Yes (GL4.6+ext)
+| SPIR-V |   Yes           |     Yes         |    Yes (SPV1.5+ext)   | Yes (SPV1.5+ext) | Yes (SPV1.5+ext)
++--------+-----------------+-----------------+-----------------------+------------------+------------------+
+
+
+ConstantBuffer, (RW)StructuredBuffer (RW)ByteAddressBuffer
+----------------------------------------------------------
+
+Each member in a `ConstantBuffer` will be emitted as `uniform` parameter.
+StructuredBuffer and ByteAddressBuffer are translated to a shader storage buffer with `readonly` layout.
+RWStructuredBuffer and RWByteAddressBuffer are translated to a shader storage buffer with `read-write` layout.
+
+
+ParameterBlock for SPIR-V target
+--------------------------------
+
+`ParameterBlock` is a Slang generic type for binding uniform parameters.
+It is similar to `ConstantBuffer` in HLSL, but `ParameterBlock` can include not only constant parameters but also descriptors such as Texture2D or StructuredBuffer.
+
+Because the Vulkan API doesn't natively support `ParameterBlock` that has a mixture of descriptors and constants, Slang emits each member as an individual parameter.
+When targeting SPIR-V, the individual parameter will be emitted as either `uniform` or `buffer`.
+
+
+SPIR-V specific Compiler options
+--------------------------------
+
+The following compiler options are specific to SPIR-V.
+
+### -emit-spirv-directly
+Generate SPIR-V output directly (default)
+It cannot be used with -emit-spirv-via-glsl
+
+### -emit-spirv-via-glsl
+Generate SPIR-V output by compiling generated GLSL with glslang
+It cannot be used with -emit-spirv-directly
+
+### -g
+Include debug information in the generated code, where possible.
+When targeting SPIR-V, this option emits [SPIR-V NonSemantic Shader DebugInfo Instructions](https://github.com/KhronosGroup/SPIRV-Registry/blob/main/nonsemantic/NonSemantic.Shader.DebugInfo.100.asciidoc).
+
+### -O<optimization-level>
+Set the optimization level.
+When targeting SPIR-V, this option applies to the downstream compiler.
+"-O0" option can help to identify issues when the problem is from the optimization step of the downstream compiler.
+Compared to the macro inlining, the specialization helps debugging with "-O0" option, because the specialization preserves the code structure much more than the macro inlining.
```

Comment:
I think we want to say "Under "-O0" option, Slang will not perform extensive inlining for all functions calls, instead it will preserve the call graph as much as possible to help with understanding the SPIRV structure and diagnosing any downstream toolchain issues."

Reply:
I will change as suggested.